pkgs/applications/misc/plover/default.nix

```diff
@@ -1,13 +1,72 @@
-{ lib, fetchFromGitHub, python3Packages, wmctrl, qtbase, mkDerivationWith }:
+{ lib, fetchFromGitHub, python310Packages, wmctrl, qtbase, mkDerivationWith }:
 
 {
   stable = throw "plover.stable was removed because it used Python 2. Use plover.dev instead."; # added 2022-06-05
 
-  dev = with python3Packages; mkDerivationWith buildPythonPackage rec {
+  dev = with python310Packages; let
+    plover_stroke = mkDerivationWith buildPythonPackage rec {
+      pname = "plover_stroke";
+      version = "1.1.0";
+
+      pyproject = true;
+
+      src = fetchFromGitHub {
+        owner = "openstenoproject";
+        repo = "plover_stroke";
+        rev = "${version}";
+        sha256 = "sha256-A75OMzmEn0VmDAvmQCp6/7uptxzwWJTwsih3kWlYioA=";
+      };
+
+      meta = with lib; {
+        description = "Stroke handling helper library for Plover";
+        license = licenses.gpl2Plus;
+        homepage = "https://github.com/openstenoproject/plover_stroke";
+        platforms = platforms.linux ++ platforms.darwin ++ platforms.windows;
+        maintainers = with maintainers; [ FirelightFlagboy ];
+      };
+
+      nativeCheckInputs = [ pytest ];
+      propagatedBuildInputs = [ setuptools ];
+
+      pythonImportsCheck = [ "plover_stroke" ];
+    };
+
+    rtf_tokenize = mkDerivationWith buildPythonPackage rec {
+      pname = "rtf_tokenize";
+      version = "1.0.0";
+
+      pyproject = true;
+
+      src = fetchFromGitHub {
+        owner = "openstenoproject";
+        repo = "rtf_tokenize";
+        rev = "${version}";
+        sha256 = "sha256-zwD2sRYTY1Kmm/Ag2hps9VRdUyQoi4zKtDPR+F52t9A=";
+      };
+
+      meta = with lib; {
+        description = "Simple RTF tokenizer";
+        license = licenses.gpl2Plus;
+        homepage = "https://github.com/openstenoproject/rtf_tokenize";
+        platforms = platforms.linux ++ platforms.darwin ++ platforms.windows;
+        maintainers = with maintainers; [ FirelightFlagboy ];
+      };
+
+      nativeCheckInputs = [ pytest ];
+      propagatedBuildInputs = [ setuptools ];
+
+      pythonImportsCheck = [
+        "rtf_tokenize"
+      ];
+    };
+  in mkDerivationWith buildPythonPackage rec {
     pname = "plover";
-    version = "4.0.0.dev10";
+    version = "4.0.0.dev12";
```

Comment:
The current pattern for `plover` on `nixpkgs` is `plover.stable` (deprecated) and `plover.dev`. 

Since it's my first contribution to `nixpkgs` I wanted to start with something simple and providing the `v4.0.0-rc.2` would have required to add a new derivation for `plover.rc` (by following the previous pattern) (but I may provide a PR for that version in the future).

Instead I've choose to provide an update for `plover.dev` even tho it's an old version (2022) to take a hands-on the PR process of `nixpkgs`.

Reply:
Ah I see, I don't think you'd need to add a new derivation `plover.rc`, I think `plover.dev` would be fine with a release candidate version -- I think the distinction is full/stable releases vs development releases, and a release candidate I would say is a development release (on its way to being a full release though).

The other way of looking at it is, I don't believe v4.0.0rc2 breaks anything new that v4.0.0dev10 or v4.0.0dev12 doesn't already break.

I.e. both have the following

> Warning: Version 4 is a major change and the configuration file it creates is not compatible with Plover 3 or earlier. Please backup your plover.cfg

and this might be the reason for `plover.stable` vs `plover.dev`, but it's not a reason to have `plover.dev` vs `plover.rc` if this makes sense.

So upgrading to v4.0.0rc2 but keeping it under `plover.dev` should be fine.

Reply:
I've added a commit to update to `4.0.0-rc.2` \(^^)/

Reply:
I agree with this, I'm perfectly happy for `plover.dev` to be an RC.  I only added the `stable`/`dev` split because `stable` didn't have any updates for a very long time.  If we consider `plover.stable` to be deprecated and we don't expect many people to be using it (the last stable release of Plover is from 2017!) then I wouldn't even mind doing away with this distinction and having the main `plover` attribute refer to 4.0.0-rc.2.